raft/log.go

```diff
@@ -1021,9 +1026,11 @@ func (l *Log) candidateLoop(closing <-chan struct{}) State {
 			l.lock()
 			l.mustSetTermIfHigher(hb.term)
 			if hb.term >= l.term {
+				l.tracef("candidateLoop: new leader: old=%d new=%d", l.leaderID, hb.leaderID)
 				l.leaderID = hb.leaderID
 			}
 			l.unlock()
+			return Follower
```

Comment:
According to the paper, candidate -> follower should only happen if `hb.term >= l.term`. See section 5.2 (and the code above).